src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -158,32 +159,46 @@ public void syncSubscription(String subscriptionId) {
   }
 
   void syncSubscriptions(String orgId, int offset, int limit) {
-    log.info("Syncing subscriptions for org={} with offset={} and limit={} ", orgId, offset, limit);
+    log.info(
+        "Syncing subscriptions for orgId={} with offset={} and limit={} ", orgId, offset, limit);
     Timer.Sample syncTime = Timer.start();
 
     int pageSize = limit + 1;
     List<Subscription> subscriptions =
         subscriptionService.getSubscriptionsByOrgId(orgId, offset, pageSize);
+    int numFetchedSubs = subscriptions.size();
+    boolean hasMore = numFetchedSubs >= pageSize;
     log.info(
-        "Fetched subscriptions of size={} for org={} from external service ",
-        subscriptions.size(),
-        orgId);
+        "Fetched numFetchedSubs={} for orgId={} from external service.", numFetchedSubs, orgId);
+
+    subscriptions =
+        subscriptions.stream().filter(this::shouldSyncSub).collect(Collectors.toUnmodifiableList());
+    int numKeptSubs = subscriptions.size();
 
-    boolean hasMore = subscriptions.size() >= pageSize;
     subscriptions.forEach(this::syncSubscription);
     if (hasMore) {
       enqueueSubscriptionSync(orgId, offset + limit, limit);
     }
     Duration syncDuration = Duration.ofNanos(syncTime.stop(syncTimer));
     log.info(
-        "Fetched and synced numSubs={} for orgId={} offset={} limit={} in subSyncedTimeMillis={}",
-        subscriptions.size(),
+        "Fetched numFetchedSubs={} and synced numSyncedSubs={} active/recent subscriptions for orgId={} offset={} limit={} in subSyncedTimeMillis={}",
+        numFetchedSubs,
+        numKeptSubs,
         orgId,
         offset,
         limit,
         syncDuration.toMillis());
   }
 
+  private boolean shouldSyncSub(Subscription sub) {
+    // Reject subs expired long ago, or subs that won't be active quite yet.
+    OffsetDateTime now = clock.now();
+    long earliestAllowedFutureStartDate = now.plusMonths(2).toEpochSecond() * 1000;
+    long latestAllowedExpiredEndDate = now.minusMonths(2).toEpochSecond() * 1000;
+    return sub.getEffectiveStartDate() < earliestAllowedFutureStartDate
+        && sub.getEffectiveEndDate() > latestAllowedExpiredEndDate;
```

Comment:
By limiting to only refreshing subscriptions that have only recently expired or are near-future-dated, this improved time-to-sync on my local machine with the test-org with lots of subs by 2x.

Reply:
1. Can you please update the StubSearchApi to populate these fields? Otherwise, I get NPE when using the stub.
2. If we are 100% sure that the API won't return `null`, we should probably mark the properties as required in the spec. This will prevent intellij (and possibly other tools, but not sonar this time?) from freaking out over possible NPE.
![image](https://user-images.githubusercontent.com/1716882/135926433-2b9fbd7d-91ed-4846-b57e-d5dc215df807.png)

3. Can you please put the cutoffs in configuration?
 a. Add Duration properties to `SubscriptionServiceProperties` for both.
 b. Add entries to `application.yaml` for both (with defaults).
 c. Add template params to `templates/rhsm-subscriptions-capacity-ingress.yml` along with corresponding pod env vars (similar to `SUBSCRIPTION_URL`).

Reply:
1. Will do!
2. I had to double-check... Turns out effectiveStartDate and effectiveEndDate *CAN* be null. I will investigate this further with the subprod team, as I don't see how that is possible... In every instance, at least one subscription product had a non-null activeStartDate and activeEndDate. In the meantime, I will allow such subs to "slip past" this filter for now, as that is how it would've happened before this code was added.
3. I debated adding this, and figured if somebody wanted it, it'd come up in the PR. And that's exactly what happened, so woot! I'll get it added.

Reply:
Revised answer:
1. It is possible for effecdtiveStartDate and effectiveEndDate to be null, but they *SHOULDN'T*. That is, if they come back null from the upstream server, then that indicates bad data and needs to be fixed. I added logging that will tell us when this happens so we can identify the subs and ask for them to be corrected upstream. When we encounter such a sub, we will skip syncing it.
2. Added unit tests to cover the null case of these dates, and to skip them if so.
3. Cutoffs are now configurable:
    a. I used Period instead of Duration, since Period is date-scale and allows for years, months, and days, whereas Duration is time-scale and allows for seconds, minutes, hours, days. Since subscription dates are dates (the fact that it is represented upstream as an exact timestamp is a by-product of the tech used, as I understand it), Period seems a better fit.
    b. added
    c. added